tldr/mcp_server.py

```
@@ -129,8 +132,27 @@ def _ensure_daemon(project: str, timeout: float = 10.0) -> None:
+            # Release lock BEFORE waiting - on Windows, holding a lock while
+            # spawning a child with start_new_session=True can cause the child
+            # to fail to start properly (lock inheritance issue)
+            if os.name == "nt":
+                try:
+                    msvcrt.locking(lock_file.fileno(), msvcrt.LK_UNLCK, 1)
+                except OSError:
+                    pass
+            else:
+                fcntl.flock(lock_file.fileno(), fcntl.LOCK_UN)
+
             subprocess.Popen(
```

Ruff (0.15.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)